### PR TITLE
RavenDB-11415 Perform a performance test on cluster-wide transactions

### PR DESCRIPTION
**Pull request by karmeli87**
still WIP

- Fixing concurrent usage of context in the cluster transaction.
- Fixing confusion between the 'raft index' and the 'previous cluster command count'
- Adding Session bench-marking.

### CODE REVIEW

**Comment by ppekrol** [comment]
@karmeli87 please check failing tests

**Comment by karmeli87** [comment]
Done, was unrelated to this specific PR, rather the `SerializedDatabaseCommands` wasn't cloned.

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
@talweiss1982 Can you check this?

File: bench/BulkInsert.Benchmark/Program.cs [hunk 1]
```diff
@@ -104,7 +106,140 @@ private string RandomString(int length)
...
 
+        private static string[] _randStr;
+        public void PerformSessionStore(string collection, long numberOfDocuments, int docsPerSession, int numberOfClients, int? sizeOfDocuments,
+            bool useSeqId = true)
+        {
+            int docSize = sizeOfDocuments == null ? _random.Next(5 * 1024 * 1024) : sizeOfDocuments.Value;
+            var sizeStr = sizeOfDocuments == null ? $"Random Size of {docSize:#,#}" : $"size of {sizeOfDocuments:#,#}";
+            Log($"About to perform Store for {numberOfDocuments:##,###} documents with {sizeStr} from {numberOfClients} clients.");
+            
+            var dummies = Math.Min(numberOfDocuments, 1024 * 1024 * 1024 / docSize + 1);
```
Comment (talweiss1982) [inline bench/BulkInsert.Benchmark/Program.cs:117]: You can use Math.Ceiling() for such calculation in the future, if 1GB%docSize == 0  then you're off by one, not an issue for this benchmark but for places where it is important you should be careful.

**Pull request merged**